## Analysis

In [1]:
import pandas as pd

## Head of Data

In [2]:
csvfile = "Resources/purchase_data.csv"
purchase_data_df = pd.read_csv(csvfile)
purchase_data_df.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

## Player Count
* There are 576 players

In [3]:
unique_players = purchase_data_df["SN"].value_counts()

total_players = len(unique_players)

total_players_df = pd.DataFrame({"Total Players": [total_players]})
total_players_df

Total Players
0            576

## Total Purchasing Analysis 
* There are 179 different games 
<hr>
* The average Price of games is \\$3.50
<hr>
* 780 games have been purchased
<hr>
* The total revenue of games is \\$2,379.77

In [4]:
# PURCHASING ANALYTICS
# creating summary data frame
unique_items = purchase_data_df["Item Name"].value_counts()

num_of_unique_items = len(unique_items)

avg_price = purchase_data_df["Price"].mean()

avg_price = round(avg_price,2)

total_purchases = purchase_data_df["Item Name"].count()

total_rev = purchase_data_df["Price"].sum()

summary_df = pd.DataFrame([
    {"Number of Unique Items": num_of_unique_items, 
     "Average Price": f"${avg_price}", 
     "Number of Purchases": total_purchases, 
     "Total Revenue": total_rev}
])
summary_df["Total Revenue"] = summary_df["Total Revenue"].map("${:,.2f}".format)
summary_df

Number of Unique Items Average Price  Number of Purchases Total Revenue
0                     179         $3.05                  780     $2,379.77

## Gender Demographics

* There are 81 female players, which makes up for 14.06% of the general player population.
<hr>
* There are 484 male players, which makes up for 84.03% of the general player population.
<hr>
* There are 11 other/non-disclosed gendered players, which makes up for 1.91% of the general player population.

In [5]:
# GENDER DEMOGRAPHIC
grouped_gender = purchase_data_df.groupby("Gender")

total_count = grouped_gender["SN"].nunique()

percent_players = total_count / total_players

gender_demographics_data = pd.DataFrame({"Total Count": total_count,
                                  "Percentage of Players": percent_players*100})
gender_demographics_data["Percentage of Players"] = gender_demographics_data["Percentage of Players"].map("{:.2f}%".format)
gender_demographics_data

Total Count Percentage of Players
Gender                                                  
Female                          81                14.06%
Male                           484                84.03%
Other / Non-Disclosed           11                 1.91%

## Purchasing Analysis for Gender
Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. by gender
* The female population purchased 113 games in total, with an average purchase price of \\$3.20 with each female player spending an average of \\$4.47 on games.
<hr>
* The male population purchased 652 games in total, with an average purchase price of \\$3.02 with each player spending an average of \\$4.07 on games.
<hr>
* The other/non-disclosed gendered population purchased 15 games in total, with an average purchase price of \\$3.35 with each player spending an average of \\$4.56 on games.

In [6]:
# PURCHASING ANALYSIS (GENDER)

gender_purchase_count = grouped_gender["Purchase ID"].count()

avg_purchase_price = grouped_gender["Price"].sum() / grouped_gender["Price"].count()

avg_total_per_person = grouped_gender["Price"].sum() / total_count

gender_data = pd.DataFrame({"Purhcase Count": gender_purchase_count,
                            "Average Purchase Price":avg_purchase_price,
                            "Avg Total Purchase per Person":avg_total_per_person})
gender_data["Average Purchase Price"] = gender_data["Average Purchase Price"].map("${:.2f}".format)
gender_data["Avg Total Purchase per Person"] = gender_data["Avg Total Purchase per Person"].map("${:.2f}".format)
gender_data.head()

Purhcase Count Average Purchase Price  \
Gender                                                         
Female                            113                  $3.20   
Male                              652                  $3.02   
Other / Non-Disclosed              15                  $3.35   

                      Avg Total Purchase per Person  
Gender                                               
Female                                        $4.47  
Male                                          $4.07  
Other / Non-Disclosed                         $4.56

## Age Demographics
* There are 17 players under 10 years old, which makes up for 2.95% of the general player population.
<hr>
* There are 22 players between 10 and 14 years old, which makes up for 3.82% of the general player population.
<hr>
* There are 107 players between 15 and 19 years old, which makes up for 18.58% of the general player population.
<hr>
* There are 258 players between 20 and 24 years old, which makes up for 44.79% of the general player population.
<hr>
* There are 77 players between 25 and 29 years old, which makes up for 13.37% of the general player population.
<hr>
* There are 52 players between 30 and 34 years old, which makes up for 9.03% of the general player population.
<hr>
* There are 31 players between 35 and 39 years old, which makes up for 5.38% of the general player population.
<hr>
* There are 11 players between 40 and 45 years old, which makes up for 1.91% of the general player population.
<hr>
*  There is 1 player over 45 years old, which makes up for 0.17% of the general player population.

In [7]:
# AGE DEMOGRAPHICS

# binning data
purchase_data1_df = purchase_data_df
bins = [0, 9, 14, 19, 24, 29, 34, 39, 44, 100]
labels = ["<10", "10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "40-44", "45+"]

purchase_data1_df["Age Dem"]=pd.cut(purchase_data1_df["Age"], bins, labels=labels)
age_group = purchase_data1_df.groupby("Age Dem")
age_count = age_group["SN"].nunique()

age_percent = age_count / total_players

age_demographics_data = pd.DataFrame({"Total Count": age_count,
                                      "Percentage of Players": age_percent*100})
age_demographics_data["Percentage of Players"] = age_demographics_data["Percentage of Players"].map("{:.2f}%".format)

age_demographics_data

Total Count Percentage of Players
Age Dem                                   
<10               17                 2.95%
10-14             22                 3.82%
15-19            107                18.58%
20-24            258                44.79%
25-29             77                13.37%
30-34             52                 9.03%
35-39             31                 5.38%
40-44             11                 1.91%
45+                1                 0.17%

## Purchasing Analysis for Age
* The players under 10 years old have purchased 23 games in total, have an average purchse price per game of \\$3.35, have purchased a total of \\$77.13 in games, and have spent an average of \\$4.54 per person.  
<hr>
* The players between 10 and 14 years old have purchased 28 games in total, have an average purchse price per game of \\$2.96, have purchased a total of \\$77.13 in games, and have spent an average of \\$3.76 per person.
<hr>
* The players between 15 and 19 years old have purchased 136 games in total, have an average purchse price per game of \\$3.04, have purchased a total of \\$82.78 in games, and have spent an average of \\$3.86 per person.
<hr>
* The players between 20 and 24 years old have purchased 365 games in total, have an average purchse price per game of \\$3.05, have purchased a total of \\$412.89 in games, and have spent an average of \\$4.32 per person.
<hr>
* The players between 25 and 29 years old have purchased 101 games in total, have an average purchse price per game of \\$2.90, have purchased a total of \\$1114.06 in games, and have spent an average of \\$3.81 per person.
<hr>
* The players between 30 and 34 years old have purchased 73 games in total, have an average purchse price per game of \\$2.93, have purchased a total of \\$293.00 in games, and have spent an average of \\$4.12 per person.
<hr>
* The players between 35 and 39 years old have purchased 41 games in total, have an average purchse price per game of \\$3.60, have purchased a total of \\$214.00 in games, and have spent an average of \\$4.76 per person.
<hr>
* The players between 40 and 44 years old have purchased 12 games in total, have an average purchse price per game of \\$3.04, have purchased a total of \\$147.67 in games, and have spent an average of \\$3.32 per person.
<hr>
* The players over 45 years old have purchased 1 games in total, have an average purchse price per game of \\$1.70, have purchased a total of \\$1.70 in games, and have spent an average of \\$1.70 per person.  

In [8]:
# PURCHASING ANALYSIS (AGE)
age_purchase_count = age_group["Purchase ID"].count()
age_total_purchase = age_group["Price"].sum()
age_avg_price = age_total_purchase / age_purchase_count
age_avg_purchase = age_total_purchase / age_count

# labeling variables
age_data = pd.DataFrame({"Purchase Count": age_purchase_count,
                         "Average Purchase Price": age_avg_price,
                         "Total Purchase Price": age_total_purchase,
                         "Avg Total Purchase per Person": age_avg_purchase})

# formatting
age_data["Total Purchase Price"] = age_data["Total Purchase Price"].map("${:.2f}".format)
age_data["Average Purchase Price"] = age_data["Average Purchase Price"].map("${:.2f}".format)
age_data["Avg Total Purchase per Person"] = age_data["Avg Total Purchase per Person"].map("${:.2f}".format)
age_data

Purchase Count Average Purchase Price Total Purchase Price  \
Age Dem                                                               
<10                  23                  $3.35               $77.13   
10-14                28                  $2.96               $82.78   
15-19               136                  $3.04              $412.89   
20-24               365                  $3.05             $1114.06   
25-29               101                  $2.90              $293.00   
30-34                73                  $2.93              $214.00   
35-39                41                  $3.60              $147.67   
40-44                12                  $3.04               $36.54   
45+                   1                  $1.70                $1.70   

        Avg Total Purchase per Person  
Age Dem                                
<10                             $4.54  
10-14                           $3.76  
15-19                           $3.86  
20-24                           $4.32  
25-29                           $3.81  
30-34                           $4.12  
35-39                           $4.76  
40-44                           $3.32  
45+                             $1.70

## Top Spenders
* The top 5 spenders are Lisosia93, Idastidru52, Chamjask73, Iral74, Iskadarya95. 
Lisosia93 has spent a total of \\$18.96 on 5 games. Her average purchase price is \\$3.79
<hr/>
Idastidru52 has spent a total of \\$15.45 on 4 games. Her average purchase price is \\$3.86
<hr/>
Chamjask73 has spent a total of \\$13.83 on 3 games. Her average purchase price is \\$4.61
<hr/>
Iral74 has spent a total of \\$13.62 on 4 games. Her average purchase price is \\$3.40
<hr/>
Iskadarya95 has spent a total of \\$13.10 on 3 games. Her average purchase price is \\$4.37

In [9]:
# TOP SPENDERS
# defining variables
grouped_spenders = purchase_data_df.groupby("SN")
purchase_count = grouped_spenders["Purchase ID"].count()

total_value = grouped_spenders["Price"].sum()

avg_purchase_price = total_value / purchase_count

spenders_df = pd.DataFrame({"Purchase Count": purchase_count,
                        "Average Purchase Price": avg_purchase_price,
                        "Total Purchase Value": total_value})

# converting column 'Total Purchase Value' to integer so sorting is more accurate
spenders_df['Total Purchase Value'] = pd.to_numeric(spenders_df['Total Purchase Value'])

# sorting values in descending order
top_spenders = spenders_df.sort_values("Total Purchase Value", ascending = False)

#formatting
top_spenders["Average Purchase Price"] = spenders_df["Average Purchase Price"].map("${:.2f}".format)
top_spenders["Total Purchase Value"] = spenders_df["Total Purchase Value"].map("${:.2f}".format)


top_spenders.head()

Purchase Count Average Purchase Price Total Purchase Value
SN                                                                     
Lisosia93                 5                  $3.79               $18.96
Idastidru52               4                  $3.86               $15.45
Chamjask73                3                  $4.61               $13.83
Iral74                    4                  $3.40               $13.62
Iskadarya95               3                  $4.37               $13.10

## Most Popular Items
The top 5 most popular games are:
* Final Critic with 13 purchases, priced at \\$4.61. The total revenue accrued from this game is \$59.99.
<hr/>
* Oathbreaker, Last Hope of the Breaking Storm with 12 purchases, priced at \$4.23. The total revenue accrued from this game is \$50.76.
<hr/>
* Fiery Glass Crusader with 9 purchases, priced at \$4.58. The total revenue accrued from this game is \$41.22.
<hr/>
* Persuasion with 9 purchases, priced at \$3.22. The total revenue accrued from this game is \$28.99.
<hr/>
* Extraction, Quickblade of Trembling Hands with 9 purchases, priced at \$3.53. The total revenue accrued from this game is \$31.77.


In [10]:
# MOST POPULAR ITEMS

item_df = purchase_data_df[["Item ID", "Item Name", "Price"]]

grouped_item = item_df.groupby(["Item ID", "Item Name"])

purchase_count = grouped_item["Item ID"].count()

total_value = grouped_item["Price"].sum()

price = grouped_item["Price"].mean()

item_summary = pd.DataFrame({"Item Price": price,
                      "Purchase Count": purchase_count,
                      "Total Purchase Value": total_value})

most_popular_df = item_summary.sort_values("Purchase Count", ascending = False)
most_popular_df["Total Purchase Value"] = most_popular_df["Total Purchase Value"].map("${:.2f}".format)
most_popular_df["Item Price"] = most_popular_df["Item Price"].map("${:.2f}".format)

most_popular_df.head()

Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
132     Persuasion                                        $3.22   
108     Extraction, Quickblade Of Trembling Hands         $3.53   

                                                      Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
132     Persuasion                                                 9   
108     Extraction, Quickblade Of Trembling Hands                  9   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
132     Persuasion                                                 $28.99  
108     Extraction, Quickblade Of Trembling Hands                  $31.77

## Most Profitable Items
The top 5 most profitable items are:
* Final Critic with 13 purchases, priced at \\$4.61. The total revenue accrued from this game is \$59.99.
<hr/>
* Oathbreaker, Last Hope of the Breaking Storm with 12 purchases, priced at \$4.23. The total revenue accrued from this game is \\$50.76.
<hr/>
* Nirvana with 9 purchases, priced at \\$4.90. The total revenue accrued from this game is \\$44.10.
<hr/>
* Fiery Glass Crusader with 9 purchases, priced at \\$4.58. The total revenue accrued from this game is \\$41.22.
<hr/>
* Singed Scalpel with 8 purchases, priced at \\$4.35. The total revenue accrued from this game is \\$34.80.


In [11]:
# MOST PROFITABLE ITEMS

# sorting values in descending order
most_profitable_df = item_summary.sort_values("Total Purchase Value", ascending = False)

most_profitable_df["Total Purchase Value"] = most_profitable_df["Total Purchase Value"].map("${:.2f}".format)
most_profitable_df["Item Price"] = most_profitable_df["Item Price"].map("${:.2f}".format)


most_profitable_df.head()

Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
82      Nirvana                                           $4.90   
145     Fiery Glass Crusader                              $4.58   
103     Singed Scalpel                                    $4.35   

                                                      Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
82      Nirvana                                                    9   
145     Fiery Glass Crusader                                       9   
103     Singed Scalpel                                             8   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
82      Nirvana                                                    $44.10  
145     Fiery Glass Crusader                                       $41.22  
103     Singed Scalpel                                             $34.80